Content.Server/GameObjects/Components/Items/RCDComponent.cs

```diff
@@ -0,0 +1,169 @@
+#pragma warning restore 649
+        public override string Name => "RCD";
+        private string _outputTile = "floor_steel";
+        private RCDmode mode = 0; //What mode are we on? Can be floors, walls, deconstruct.
+        private Array modes = Enum.GetValues(typeof(RCDmode));
+        private int ammo = 5; //How much "ammo" we have left. You can refille this with RCD ammo.
+
+        ///Enum to store the different mode states for clarity.
+        private enum RCDmode
+        {
+            Floors,
+            Walls,
+            Deconstruct
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _outputTile, "output", "floor_steel");
+        }
+
+
+        ///Method called when the RCD is clicked in-hand, this will swap the RCD's mode from "floors" to "walls".
+
+        public bool UseEntity(UseEntityEventArgs eventArgs)
+        {
+            SwapMode(eventArgs);
+            return true;
+        }
+
+        /**
+         *
+         * Method to allow the user to swap the mode of the RCD by clicking it in hand, the actual in hand clicking bit is done over on UseEntity()
+         *
+         * @param UseEntityEventArgs = The entity which triggered this method call, used to know where to play the "click" sound.
+         *
+         */
+
+        public void SwapMode(UseEntityEventArgs eventArgs)
+        {
+            _entitySystemManager.GetEntitySystem<AudioSystem>().Play("/Audio/items/genhit.ogg", Owner);
+            int mode = (int) this.mode; //Firstly, cast our RCDmode mode to an int (enums are backed by ints anyway by default)
+            mode = (++mode) % modes.Length; //Then, do a rollover on the value so it doesnt hit an invalid state
+            this.mode = (RCDmode) mode; //Finally, cast the newly acquired int mode to an RCDmode so we can use it.
+            switch (this.mode)
+            {
+                case RCDmode.Floors:
+                    _outputTile = "floor_steel";
+                    break;
+                case RCDmode.Walls:
+                    _outputTile = "base_wall";
+                    break;
+                case RCDmode.Deconstruct:
+                    _outputTile = "space";
+                    break;
+            }
+            _serverNotifyManager.PopupMessage(Owner, eventArgs.User, "The RCD is now set to "+this.mode+" mode."); //Prints an overhead message above the RCD
+        }
+
+        /**
+         *
+         * Method called when the user examines this object, it'll simply add the mode that it's in to the object's description
+         * @params message = The original message from examining, like ..() in BYOND's examine
+         *
+         */
+
+        void IExamine.Examine(FormattedMessage message)
+        {
+            message.AddMarkup(Loc.GetString("It's currently in "+mode+" mode, and holds "+this.ammo+" charges."));
```

Comment:
Keep in mind the first parameter of "GetString" will be translated. If it can have multiple forms (e.g. "in floor mode, and olds 43 charges"), that's more translations to do (one for every mode and every ammunition amount possible.

Use the ability of Log.GetString to work as a [C# string formatter](https://docs.microsoft.com/fr-fr/dotnet/api/system.string.format?view=netcore-3.1) (a bit like printf in C, it you are familiar with it).

Use this to add "parameters" to the string, that will not need to be translated:
```c#
Loc.GetString("It's currently placing {0}, and holds {1} charges.", Loc.GetParticularString("RCDMode", mode), this.ammo);
```

Reply:
That's pretty sweet, thanks!